torchao/quantization/prototype/qat/utils.py

```diff
@@ -143,3 +179,60 @@ def _choose_qparams_per_token_asymmetric(
     zero_point = torch.clamp(zero_point, qmin, qmax).round()
 
     return scale.to(scales_precision), zero_point.to(zero_points_precision)
+
+def _forward_pre_hook_handler(
+    mod: torch.nn.Linear,
+    prehook: Callable,
+    handler: torch.utils.hooks.RemovableHandle,
+):
+    """
+    Store a 2-tuple (prehook function, handler) as an attribute on the given linear module.
+    """
+    setattr(mod, _QAT_LINEAR_SUBCLASS_INPUT_PREHOOK, (prehook, handler))
```

Comment:
why do you need to stash the hooks / handles themselves onto the module? (when you register an nn module hook, it will automatically be stored somewhere on the module)

Reply:
oh I see, you want the user to be able to enable/disable the entire flow on a given layer.